### DISCUSSION TITLE
Hyprland process taking 85% CPU in VM

### DISCUSSION THREAD

**Discussion by RedBearAK** [JaKooLit/Hyprland-Dots#86] [General]
FYI, I think I figured out why Hyprland is so sluggish in my Boxes virtual machine. The "Hyprland" process is constantly using about 85% of the CPU, starting shortly after I log in. I give my VMs 4GB of RAM and access to all of the CPU cores I have on a Ryzen 3700u (4 cores, 8 threads). They have plenty of resources. 

There's no other process using any significant CPU time. Nothing obvious showing up in the systemd journal. Not sure how to troubleshoot it any further. 

Same thing happens when I shut down Boxes and boot the VM with Virt-Manager. I think they are both just front-ends to libvirt and QEMU/KVM, so I didn't expect any difference. But I definitely don't normally see this with any other Linux distro or desktop environment, even ones that don't work well in VMs (like elementary OS). 

Logging out of Hyprland and back into GNOME on the same Fedora 39 VM results in CPU usage going down to about 2%.

**Comment by RedBearAK** [comment]
I kept checking the Discussions tab and this thread wasn't showing up for some reason. 

What's weird now is after getting the CPU usage under control with the 3D acceleration and OpenGL being enabled, when I actually try to do things in Hyprland (even working with a terminal app) it seems even more sluggish than it was when 3D acceleration was disabled. Maybe it's being held back by not having a dedicated GPU. I only have integrated Vega 10 graphics to work with. 

There's a delay between all actions, even moving the mouse cursor around. On the other hand, window animations like when the HINT window appears are much smoother. 

Hyprland is an enigma.

**Reply by JaKooLit** [reply]
thats why hyprland is not old generations friendly since it requires an actual 3d acceleration.

On any virtual environment, the 3d acceleration is only software..

Thats why any systems older than 5 yrs kinda struggle on my observation or 5 generations older kinda struggle
